config/config.go

```diff
@@ -18,9 +18,24 @@ type (
 
 	// Scanner contains the configuration for the host scanner.
 	Scanner struct {
-		Threads             int           `yaml:"threads,omitempty"`
-		Timeout             time.Duration `yaml:"timeout,omitempty"`
-		MaxLastScan         time.Duration `yaml:"maxLastScan,omitempty"`
+		// BatchSize represents the maximum number of hosts we will
```

Comment:
The docstrings are great, I like the field names less. 

`BatchSize` -> `NumThreads`: since every unscanned host gets its own goroutine, threads cover the load better
`Timeout` -> `ScanTimeout`: doesn't really matter
`CheckAgainDelay` -> `ScanFrequency`: doesn't really matter
`MaxLastScan` -> `ScanInterval`: I didn't really get the `Max` part? There's no `LastScan` field and it's not even the max possible value because we have an exp backoff on failures. It's different from `MinLastAnnouncement` which clearly gets used as a cutoff for the `LastAnnouncement` column.

Another one I noticed is `FailedInteractionsStreak`, which we would usually call `ConsecutiveFailedScans`, which is also how you describe it in the docstring of `MaxLastScan`.